2_process/src/filter_wqp_data.R

```diff
@@ -54,37 +80,50 @@ subset_wqp_SC_data <- function(filtered_data,omit_dups = TRUE){
     # that we can use to look for duplicated date-times
-    mutate(ActivityStartDateTime = na_if(ActivityStartDateTime, "NA")) %>%
+    # Note that "NA" values were handled in the updated data harmonization scripts, so 
+    # commenting out this line for now.
+    #mutate(ActivityStartDateTime = na_if(ActivityStartDateTime, "NA")) %>%
     mutate(ActivityStartDateTime_filled = if_else(is.na(ActivityStartDateTime),
-                                           paste(ActivityStartDate,"12:00:00",sep=" "),
-                                           ActivityStartDateTime))
+                                           paste(ActivityStartDate, "12:00:00", sep=" "),
+                                           as.character(ActivityStartDateTime)))
   
-if(omit_dups == "TRUE"){
-  # When duplicate observations exist for a unique combination of [site name & date-time &
-  # geographic location & collecting organization], select one observation based on 
-  # the `param` attribute
-  SC_data_subset_omit_dups <- SC_data_subset %>%
-    group_by(MonitoringLocationIdentifier,
-             ActivityStartDateTime_filled, 
-             OrganizationIdentifier, 
-             LongitudeMeasure, LatitudeMeasure) %>% 
-    # Preferentially retain samples with param equals "Specific conductance, lab" first 
-    # and "Specific conductance, field, mean" last
-    arrange(match(param, c("Specific conductance, lab",
-                           "Specific conductance",
-                           "Specific conductance, field",
-                           "Specific conductance, field, mean"))) %>%
-    slice(1) %>%
-    ungroup()
+  if(omit_duplicates){
+    # When duplicate observations exist for a unique combination of 
+    # [site name & date-time & geographic location & collecting organization], 
+    # select one observation based on the `param` attribute
+    SC_data_subset_out <- SC_data_subset %>%
+      group_by(MonitoringLocationIdentifier,
+               ActivityStartDateTime_filled, 
+               OrganizationIdentifier, 
+               LongitudeMeasure, LatitudeMeasure) %>% 
+      # Preferentially retain samples with param equals "Specific conductance, lab" 
+      # first and "Specific conductance, field, mean" last.
+      arrange(match(param, c("Specific conductance, lab",
+                             "Specific conductance",
+                             "Specific conductance, field",
+                             "Specific conductance, field, mean"))) %>%
+      mutate(n_duplicated = n(),
+             dup_number = seq(n_duplicated),
+             flag_duplicate_drop = n_duplicated > 1 & dup_number != 1) %>% 
+      filter(flag_duplicate_drop == FALSE) %>%
+      ungroup() %>%
+      select(-c(n_duplicated, dup_number, flag_duplicate_drop)) %>%
+      # arrange all rows to maintain consistency in row order across users/machines
+      arrange(across(everything()))
```

Comment:
Yeah, there is some duplicated code in here. I also considered just omitting the `p2_wqp_SC_dups` and `subset_wqp_SC_dups()` because we're not using this information elsewhere in the pipeline.

Reply:
In case we want to keep the target `p2_wqp_SC_dups` I opted to simplify `subset_wqp_SC_dups()` in https://github.com/USGS-R/drb-inland-salinity-ml/pull/180/commits/a53f47fc7d156e18edacca1a0b54e6fd70e3b4bb